addons/stock/models/stock_quant.py

```diff
@@ -281,6 +277,17 @@ def _update_reserved_quantity(self, product_id, location_id, quantity, lot_id=No
                 break
         return reserved_quants
 
+    @api.model
+    def _unlink_zero_quants(self):
+        """ _update_available_quantity may leave quants with no
+        quantity and no reserved_quantity. It used to directly unlink
+        these zero quants but this proved to hurt the performance as
+        this method is often called in batch and each unlink invalidate
+        the cache. We defer the calls to unlink in this method.
+        """
+        query = "DELETE FROM stock_quant WHERE round(quantity::numeric, 5) = 0 AND round(reserved_quantity::numeric, 5) = 0;"
```

Comment:
@odony not sure how we could justify the use of 5 here. do you think using the product unit of measure digits field would be ok?

Reply:
You could use the digital precision and double it for safety. With 3 digits of precision, it makes sense to discard quantities smaller than 10^-6, i.e. 0.000001.

Do we have an official "maximum supported decimal precision" for WMS? I think after 5 digits of precision the rounding errors will become significant enough that it makes sense to just use a smaller reference UoM from the start.

Based on this you could probably save the extra SELECT and just use 10 or 12 here.

PS: Obligatory xkcd https://xkcd.com/221 ;-)

Reply:
You might need to take extra care if you use the digital precision. Nothing prevents the user to change it to a low value (e.g. 0), run a few tests then set it back to its original value. This might have unexpected results. Maybe set a minimum value (6, which would be twice the default decimal precision).